tempodb/search/backend_search_block.go

```diff
@@ -76,20 +76,17 @@ func (s *SearchDataIterator) Next(_ context.Context) (common.ID, []byte, error)
 
 	currentRecord := s.records[s.currentIndex]
 
-	// resize/extend buffer
-	if cap(s.buffer) < int(currentRecord.Length) {
-		s.buffer = make([]byte, currentRecord.Length)
-	}
-	s.buffer = s.buffer[:currentRecord.Length]
-
-	_, err := s.file.ReadAt(s.buffer, int64(currentRecord.Start))
+	// Use unique buffer that can be returned to the caller.
+	// This is primarily for DedupingIterator which uses 2 buffers at once.
+	buffer := make([]byte, currentRecord.Length)
```

Comment:
Should we add a test around this? Might be easy enough to do?

Reply:
Yes tried, but unfortunately wasn't able to reproduce the issue in a go test, even after changing `newBackendSearchBlockWithTraces` to purposefully write duplicate data.  I think there are other factors at play, like maybe the 2nd buffer read has to be smaller than the 1st, so that it only overwrites part of the buffer.

Reply:
oh :( alright yeah we can skip this, the comment was a good idea.